client/src/components/WorkoutSession/hooks/useActiveExercise.tsx

```diff
@@ -0,0 +1,51 @@
+import { ChangeEvent, InputHTMLAttributes, useState } from "react";
+import { useSetRecoilState } from "recoil";
+import { ID } from "../../../types/shared/id.types";
+import { ExerciseScheme } from "../../../types/shared/session.types";
+import { activeWorkoutState } from "../state/workout-state";
+import { isScheme } from "../types/is-scheme";
+
+export default function useActiveExercise(
+	defaultWarmupScheme: Partial<ExerciseScheme>,
```

Comment:
`defaultWarmupScheme` has `weight_unit`, `exercise_id`, and `is_warmup`.

`weight_unit` is passed as an argument. If it didn't have to be, we could define this `defaultWarmupScheme` inside `useActiveExercise()` instead of passing it from `ActiveExercise`. I'd prefer this.

Maybe consider making `weight_unit` a user-specified setting (which is way better for UX anyway), and retrieving it from localStorage (or through a `useUserSettings` hook that interacts with a global piece of state, that in turn interacts with localStorage.